tests/test_config.py

```diff
@@ -183,6 +184,123 @@ def test_get(self):
                 self.assertIsInstance(server_config.auth, tuple)
 
 
+class ReprTestCase(TestCase):
+    """Test method ``nailgun.config.BaseServerConfig.__repr__``."""
+
+    def test_bsc_v1(self):
+        """Test :class:`nailgun.config.BaseServerConfig`.
+
+        Assert that ``__repr__`` works correctly when ``url`` is specified.
+
+        """
+        target = "nailgun.config.BaseServerConfig(url='bogus')"
+        self.assertEqual(target, repr(BaseServerConfig('bogus')))
+        import nailgun  # noqa pylint:disable=unused-variable
+        # pylint:disable=eval-used
+        self.assertEqual(target, repr(eval(repr(BaseServerConfig('bogus')))))
```

Comment:
This is asserting that you can get the representation and just paste it in a python shell for example?

Also I'm interested more on the reason to do this kind of assertion :smile: